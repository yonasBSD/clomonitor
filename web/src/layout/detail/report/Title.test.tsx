import { render, screen } from '@testing-library/react';
import { vi } from 'vitest';

import Title from './Title';

const defaultProps = {
  title: 'title',
  icon: <>icon</>,
};

describe('Title', () => {
  afterEach(() => {
    vi.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<Title {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders option', () => {
      render(<Title {...defaultProps} />);

      expect(screen.getByText('title')).toBeInTheDocument();
      expect(screen.getByText('icon')).toBeInTheDocument();
    });
  });
});
