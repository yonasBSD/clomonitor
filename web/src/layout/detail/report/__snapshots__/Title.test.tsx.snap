// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Title > creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-row align-items-center undefined"
  >
    icon
    <div
      class="ms-2 fw-bold"
    >
      title
    </div>
  </div>
</DocumentFragment>
`;
